### PR TITLE
KEP-1959: add prod readiness review for alpha

### PR DESCRIPTION
**Pull request by andrewsykim**
Add alpha prod readiness review for KEP-1959

### CODE REVIEW

File: keps/sig-cloud-provider/1959-service-lb-class-field/README.md [hunk 1]
```diff
@@ -132,16 +139,18 @@ type ServiceSpec struct {
...
 	// implementation (e.g. cloud providers) should ignore Services that set this field.
 	// +optional
-	LoadBalancerClassName string `json:"loadBalancerClass,omitempty"`
+	LoadBalancerClass string `json:"loadBalancerClass,omitempty"`
 }
 ```
 
 * `loadBalancerClass` will be immutable only when the Service type is `LoadBalancer`, this way existing and future implementations
-do not have to worry about handling Services that change the class name. The class name is mutable and can be cleared when the
+do not have to worry about handling Services that change the class name. The class name is mutable and must be cleared when the
```
Comment (thockin) [inline keps/sig-cloud-provider/1959-service-lb-class-field/README.md:147]: I'm confused:

"`loadBalancerClass` will be immutable only when the Service type is `LoadBalancer` ... The class name is mutable and must be cleared when..."

Is it mutable or immutable?

File: keps/sig-cloud-provider/1959-service-lb-class-field/README.md [hunk 2]
```diff
@@ -181,6 +190,166 @@ has the field set already. This ensures apiserver can handle the new field on do
...
+_For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them._
+
+_For beta, this section is required: reviewers must answer these questions._
+
+_For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field._
+
+* **Will enabling / using this feature result in any new API calls?**
+  Describe them, providing:
```
Comment (thockin) [inline keps/sig-cloud-provider/1959-service-lb-class-field/README.md:299]: template text

File: keps/sig-cloud-provider/1959-service-lb-class-field/README.md [hunk 3]
```diff
@@ -132,16 +139,18 @@ type ServiceSpec struct {
...
+do not have to worry about handling Services that change the class name. The class name is mutable and must be cleared when the
 type changes.
 * `loadBalancerClass` will be validated against label-style format.
 * the `loadBalancerClass` field will be feature gated. The field will be dropped during API strategy unless
 the feature gate is enabled.
+* all external implementations of Service LoadBalancer using a non-empty class name should use the finalizer `service.kubernetes.io/load-balancer-cleanup`
+to ensure proper garbage collection of external resources.
 
 Required updates to service controller:
 * if the class field is NOT set for a Service, allow the cloud provider to reconcile the load balancer.
```
Comment (johnbelamaric) [inline keps/sig-cloud-provider/1959-service-lb-class-field/README.md:156]: line below (157) says "annotation", but rest of the KEP says it's a field

168s/propoerly/properly/

File: keps/sig-cloud-provider/1959-service-lb-class-field/README.md [hunk 4]
```diff
@@ -181,6 +190,166 @@ has the field set already. This ensures apiserver can handle the new field on do
...
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: ServiceLoadBalancerClass
+    - Components depending on the feature gate: kube-apiserver
```
Comment (johnbelamaric) [inline keps/sig-cloud-provider/1959-service-lb-class-field/README.md:202]: controller-manager?

File: keps/sig-cloud-provider/1959-service-lb-class-field/README.md (hunk 2 above)
Comment (johnbelamaric) [inline keps/sig-cloud-provider/1959-service-lb-class-field/README.md:299]: remove this line

**Review by johnbelamaric — COMMENTED** [review]
Couple minor things, otherwise LGTM

File: keps/sig-cloud-provider/1959-service-lb-class-field/README.md (hunk 1 above)
Comment (andrewsykim) [reply keps/sig-cloud-provider/1959-service-lb-class-field/README.md:147]: oops let me clarify.

this meant to say that the field is only mutable when the type changes from LoadBalancer and it should be cleared when that happens

File: keps/sig-cloud-provider/1959-service-lb-class-field/README.md (hunk 3 above)
Comment (andrewsykim) [reply keps/sig-cloud-provider/1959-service-lb-class-field/README.md:156]: Thanks, left-over when this was initially proposing annotations

Comment (johnbelamaric) [reply keps/sig-cloud-provider/1959-service-lb-class-field/README.md:156]: still have to fix line 168 (nit)

File: keps/sig-cloud-provider/1959-service-lb-class-field/README.md [hunk 5]
```diff
@@ -181,6 +190,165 @@ has the field set already. This ensures apiserver can handle the new field on do
...
 handle enablement of this feature if a new apiserver enabled it.
 
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [X] Feature gate (also fill in values in `kep.yaml`)
```
Comment (johnbelamaric) [inline keps/sig-cloud-provider/1959-service-lb-class-field/README.md:200]: Also put these in `kep.yaml`

**Comment by johnbelamaric** [comment]
Couple nits for PRR, otherwise good.

File: keps/sig-cloud-provider/1959-service-lb-class-field/README.md (hunk 5 above)
Comment (andrewsykim) [reply keps/sig-cloud-provider/1959-service-lb-class-field/README.md:200]: done

**Review by thockin — COMMENTED** [review]
I am happy with the body change, leaving PRR to PRR-ers.

/lgtm

**Pull request merged**